activity/core.go

```diff
@@ -82,6 +84,27 @@ func Record(ctx context.Context, msg string, a ...interface{}) {
 	journal.Record(msg, a...)
 }
 
+// Warnf writes a new entry to the journal identified by the ID at `activity.JournalKey` in the
+// provided context at WARN level. It also writes to the server logs at the same level. If no ID
+// is registered, the entry is written to the server logs at the info level. If the ID is an empty
+// string - which can happen when the JournalID header is missing from an API call - it uses the
+// ID 'dead-letter-office'.
+func Warnf(ctx context.Context, msg string, a ...interface{}) {
```

Comment:
I think a tip on when to use `Warnf` vs. `Record` would be useful (e.g. what type of errors would a Warnf make sense vs. a Record), because that isn't quite clear to me.